chipsec/hal/uefi_common.py

```diff
@@ -623,63 +623,128 @@ def NextFwVolume(buffer, off = 0):
             fof += 0x2C
     return res
 
+EFI_FIRMWARE_VOLUME_HEADER = "<16sIHH8sQIIHHHBB" 
+EFI_FV_BLOCK_MAP_ENTRY = "<II" 
+FFS_ATTRIB_LARGE_FILE         = 0x01
+  
+def GetFvHeader(buffer, off = 0):
+    header_size = struct.calcsize(EFI_FIRMWARE_VOLUME_HEADER) + struct.calcsize(EFI_FV_BLOCK_MAP_ENTRY)
+    if (len(buffer) < header_size):
+        return (0,0,0)
+    size = 0
+    fof = off + struct.calcsize(EFI_FIRMWARE_VOLUME_HEADER)
+    ZeroVector, FileSystemGuid0, FileSystemGuid1,FileSystemGuid2,FileSystemGuid3, \
+    FvLength, Signature, Attributes, HeaderLength, Checksum, ExtHeaderOffset,    \
+    Reserved, Revision = struct.unpack(EFI_FIRMWARE_VOLUME_HEADER, buffer[off:off+struct.calcsize(EFI_FIRMWARE_VOLUME_HEADER)])
+    numblocks,lenblock = struct.unpack(EFI_FV_BLOCK_MAP_ENTRY,buffer[fof:fof+struct.calcsize(EFI_FV_BLOCK_MAP_ENTRY)])
+    '''
+    print "\nFV volume offset: 0x%08X" % fof
+    print "\tFvLength:         0x%08X" % FvLength
+    print "\tAttributes:       0x%08X" % Attributes
+    print "\tHeaderLength:     0x%04X" % HeaderLength
+    print "\tChecksum:         0x%04X" % Checksum
+    print "\tRevision:         0x%02X" % Revision
+    print "\tExtHeaderOffset:  0x%02X" % ExtHeaderOffset
+    print "\tReserved:         0x%02X" % Reserved
+    print "FFS Guid:     %s" % guid_str(FileSystemGuid0, FileSystemGuid1,FileSystemGuid2, FileSystemGuid3)
+    '''
+    while not (numblocks == 0 and lenblock == 0):
+        fof += struct.calcsize(EFI_FV_BLOCK_MAP_ENTRY)
+        if (fof + struct.calcsize(EFI_FV_BLOCK_MAP_ENTRY)) >= len(buffer):
```

Comment:
Would it be better to just calculate the size once and reuse it.  We seem to be calling calcsize in a number of places for the same data structure.